File: binary-search/count-element-occurence.cpp
int bisearch(const vector<int> &A, int B, int l, int r){
    int L=0,R=0,k=0,m;
    if(l<=r){
        m=(l+r)/2;
        if(A[m]==B) k++;
        L=bisearch(A,B,l,m-1);
        R=bisearch(A,B,m+1,r);
    }
    return L+R+k;
}

int Solution::findCount(const vector<int> &A, int B) {
    return bisearch(A,B,0,A.size()-1);
}

File: binary-search/count-element-occurence.py
class Solution:
    def binary_search(self, A, key, left=True):
        low = 0
        high = len(A) - 1
        result = -1

        while low <= high:
            mid = low + (high - low)/2
            if A[mid] == key:
                result = mid
                if left:
                    high = mid - 1
                else:
                    low = mid + 1
            elif A[mid] > key: high = mid - 1
            else: low = mid + 1

        return result

    def findCount(self, A, B):
        left = self.binary_search(A, B)
        right = self.binary_search(A, B, False)
        if left == -1: return 0

        return right - left + 1

